src/app/bundles/bundle-routing.module.ts

```diff
@@ -1,24 +1,31 @@
 import { NgModule } from '@angular/core';
-import { Routes, RouterModule } from '@angular/router';
+import { RouterModule, Routes } from '@angular/router';
 
 import { BundlePageComponent } from './bundle-page/bundle-page.component';
 import { BundleDetailsComponent } from './bundle-details/bundle-details.component';
 import { DocumentSelectionComponent } from './document-selection/document-selection.component';
+import { BundlePageGuard } from './bundle-page/bundle-page.guard';
 
 const routes: Routes = [
-  {path: 'detail/:id', component: BundleDetailsComponent},
-  {path: 'page/:id', component: BundlePageComponent},
-  {path: 'import/:id', component: DocumentSelectionComponent},
-  {path: '', redirectTo: 'page/', pathMatch: 'full'},
+  { path: 'detail/:id', component: BundleDetailsComponent },
+  {
+    path: 'page/:id',
+    component: BundlePageComponent,
+    canActivate: [BundlePageGuard]
+  },
+  { path: 'import/:id', component: DocumentSelectionComponent },
+  { path: '', redirectTo: 'page/', pathMatch: 'full' },
```

Comment:
might need to redirect to another page where it has the listen of bundles (if multiple can be created)